### PR TITLE
Fixes #37167 - Subscription entitlement report template

### PR DESCRIPTION
**Pull request by sjha4**
1. Remove subscription entitlement report
2. Update Subscription - general report to also show subscription expiration dates including expiry filter.

To test:

1. Run db:migrate and db:seed
2. Ensure that the Subscription - Entitlement Report is deleted.
3. Run the Subscription - General Report and ensure you see subscription expiration filter before generating report and field on the generated report.

### CODE REVIEW

File: db/migrate/20240215150131_remove_subscription_entitlements_report.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class RemoveSubscriptionEntitlementsReport < ActiveRecord::Migration[6.1]
+  def change
```
Comment (jeremylenz) [inline db/migrate/20240215150131_remove_subscription_entitlements_report.rb:2]: This should be `def up` since there's no way for Rails to reverse it..

Comment (sjha4) [reply db/migrate/20240215150131_remove_subscription_entitlements_report.rb:2]: I am able to revert it with `rollback`.Think Rails is able to manage with `change`. Did you want the same with an empty down method?

Comment (jeremylenz) [reply db/migrate/20240215150131_remove_subscription_entitlements_report.rb:2]: I doubt `rollback` recreates your deleted report and template inputs, though?

Comment (sjha4) [reply db/migrate/20240215150131_remove_subscription_entitlements_report.rb:2]: No it doesn't..It acts like an empty down method..

Comment (jeremylenz) [reply db/migrate/20240215150131_remove_subscription_entitlements_report.rb:2]: Okay. `def up` might make that a bit more clear, but if it doesn't error I'm fine with it either way. :+1:

File: app/views/unattended/report_templates/subscription_-_general_report.erb [hunk 2]
```diff
@@ -8,24 +8,36 @@ template_inputs:
   description: Search query to filter subscription entitlements. E.g. `quantity >
     0` to exclude subscriptions with unlimited entitlements. Keep unset for all subscriptions.
   advanced: false
+- name: Days from Now
+  required: true
+  input_type: user
+  description: "Show subscriptions expiring in <= this many days.  Select 'no limit' to show all entitlements."
```
Comment (jeremylenz) [inline app/views/unattended/report_templates/subscription_-_general_report.erb]: We no longer care about "entitlements"..

```suggestion
  description: "Show subscriptions expiring in <= this many days.  Select 'no limit' to show all subscriptions."
```

**Review by jeremylenz — COMMENTED** [review]
Tested with the Katello PR and works as advertised. :+1:  Just one nit below

File: app/views/unattended/report_templates/subscription_-_general_report.erb (hunk 2 above)
Comment (sjha4) [reply app/views/unattended/report_templates/subscription_-_general_report.erb]: Weird thing going on where I have updated the branch on my fork but the force push doesn't reflect on the PR.. 😕

Comment (sjha4) [reply app/views/unattended/report_templates/subscription_-_general_report.erb]: First time I have seen the "processing updates" spinner.. 😄 
Screenshot from 2024-03-01 09-31-25

Comment (sjha4) [reply app/views/unattended/report_templates/subscription_-_general_report.erb]: All good now..

**Review by jeremylenz — APPROVED** [review]

**Pull request merged**